controllers/managedocs_controller.go

```diff
@@ -311,10 +333,24 @@ func (r *ManagedOCSReconciler) SetupWithManager(mgr ctrl.Manager, ctrlOptions *c
 			&source.Kind{Type: &opv1a1.ClusterServiceVersion{}},
 			enqueueManangedOCSRequest,
 			csvPredicates,
-		).
+		)
+
+	err = r.UnrestrictedClient.Get(ctx, client.ObjectKey{Name: "egressnetworkpolicies.network.openshift.io", Namespace: ""}, egreeNetworkPolicyCRD)
```

Comment:
I don't think that we should couple the concern of setting up the reconciliation with figuring out which type of ingress policy we need to use. Please move the check outside and pass that information here via an argument.

Reply:
done